static/src/javascripts/projects/common/modules/navigation/profile.js

```diff
@@ -0,0 +1,92 @@
+// @flow
+
+import bonzo from 'bonzo';
+import fastdom from 'fastdom';
+import { getUserFromCookie } from 'common/modules/identity/api';
+
+const CONFIG = {
+    classes: {
+        container: 'js-profile-nav',
+        content: 'js-profile-info',
+        popup: 'js-profile-popup',
+        register: 'js-profile-register',
+        commentActivity: 'js-comment-activity',
+        action: 'brand-bar__item--action',
+    },
+};
+
+class Profile {
+    dom: {
+        [key: string]: ?HTMLElement,
+    };
+    opts: {
+        url: string,
+    };
+
+    constructor(options: Object): void {
+        const opts = {
+            url: 'https://profile.theguardian.com',
+        };
+
+        this.opts = Object.assign({}, opts, options);
+
+        this.dom = {};
+
+        if (document.body) {
```

Comment:
Infuriatingly forced by `flow` to wrap each assignment to `this.dom` in a check of `document.body`. If I attempted to put them within the block of a single `if` condition it complained, and it didn't like it if assigned `document.body` to a variable either that I could use on each check, I had to explicitly use `document.body`.